### PR TITLE
switching to use better check for args

### PR DESCRIPTION
**Pull request by babineaum**

### CODE REVIEW

**Comment by babineaum** [comment]
this might be redundant? if 'file' in args and args['file']

**Comment by dx-pbuckley** [comment]
I don't think `if 'file' in args and args['file']` is redundant. The first condition checks for whether there is a key called `file` in the `args` dict. If not, we don't descend into the following logic. The second condition checks that `args['file']` is not `False`. If not, we don't descend into the following logic. Both conditions (the key exists AND the key is not False) must eval to true in order to descend into the following logic.

File: github_pr.py [hunk 1]
```diff
@@ -215,29 +215,29 @@ def main():
...
+    elif 'action' in args and args['action'] == 'delete':
         github_delete_pr(**args)
 
-    elif args['action'] == 'update':
+    elif 'action' in args and args['action'] == 'update':
         github_update_pr(**args)
 
     gh = Github(args['token'])
-    if not args['numberonly'] and not args['noratelimit']:
+    if 'numberonly' not in args and not args['numberonly'] and not args['noratelimit']:
```
Comment (cchristous) [inline github_pr.py]: this should be 
```python
if (('numberonly' in args) and args['numberonly']) or (('noratelimit' in args) and args['noratelimit']):
```
you missed a check for ```'noratelimit' in args```. Also, De Morgan's Law for readability

**Comment by babineaum** [comment]
ah right

File: github_pr.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def _check_required_fields(required, **args):
 
 
 def _print_prs(prs, **args):
```
Comment (cchristous) [inline github_pr.py]: really not a fan of how ```**args``` is used excessively in this module. Not really in the scope of this PR...

Comment (babineaum) [inline github_pr.py]: me 2, we can address in the next one

**Comment by babineaum** [comment]
fixed that one i missed, and made it more readable. should be good to ship. tested too

File: github_pr.py [hunk 3]
```diff
@@ -215,31 +215,30 @@ def main():
...
+    elif 'action' in args and args['action'] == 'delete':
         github_delete_pr(**args)
 
-    elif args['action'] == 'update':
+    elif 'action' in args and args['action'] == 'update':
         github_update_pr(**args)
 
     gh = Github(args['token'])
-    if not args['numberonly'] and not args['noratelimit']:
+    if (('numberonly' not in args) and not args['numberonly']) and (('noratelimit' in args) and not args['noratelimit']):
```
Comment (cchristous) [inline github_pr.py]: please remove all of the ```not```s and change the middle ```and``` to an ```or```

Comment (cchristous) [inline github_pr.py]: also, in the current implementation there are bugs. you are checking ```'numberonly' not in args``` and if it is not in args then you try to get the value of ```args['numberonly']``` yielding an exception

**Comment by babineaum** [comment]
Noticed one more "if arg in args and not args['arg'] "

so that should be all set now. thanks @cchristous

File: github_pr.py [hunk 4]
```diff
@@ -215,31 +215,30 @@ def main():
...
+    elif 'action' in args and args['action'] == 'delete':
         github_delete_pr(**args)
 
-    elif args['action'] == 'update':
+    elif 'action' in args and args['action'] == 'update':
         github_update_pr(**args)
 
     gh = Github(args['token'])
-    if not args['numberonly'] and not args['noratelimit']:
+    if (('numberonly' in args) and not args['numberonly']) or (('noratelimit' in args) and not args['noratelimit']):
```
Comment (cchristous) [inline github_pr.py]: I don't think this is the logic you want.
you previously had ```not args['numberonly'] and not args['noratelimit']```
now you have ```not args['numberonly'] or not args['noratelimit']```
I believe you want ```args['numberonly'] or args['noratelimit']```

Comment (babineaum) [inline github_pr.py]: i disagree, i want to check if numberonly is in args and if it is set to false then print the rate limit message. Also i want to check noratelimit in args and if its set to false then print the rate limit message.

Comment (cchristous) [inline github_pr.py]: okay, so if either of them is false then you want to print the message. I just wanted to make it clear that this was a change in behavior from the previous logic.

Comment (babineaum) [inline github_pr.py]: yes, if either are false i want to print that message. thanks for double checking, i think i made it confusing cause i reversed it twice. but you caught my earlier transgression.

**Comment by dx-pbuckley** [comment]
:diamond_shape_with_a_dot_inside: (a.k.a. :shipit: )

**Comment by babineaum** [comment]
ok "Joe Diamond Quimby"....

**Pull request merged**